src/tests/ut/reserve_queue/reserve_queue_test.cpp

```diff
@@ -0,0 +1,412 @@
+    const auto doctor_queued_patients_after = sut_.awaitingPatients(doctors_[0]);
+
+    EXPECT_NE(doctor_queued_patients_before, doctor_queued_patients_after);
+    EXPECT_THAT(doctor_queued_patients_after, ElementsAre(patients_[3], patients_[2], patients_[1]));
+}
+
+TEST_F(RemoveAwaitingPatientShould, RemoveDoctorKeyIfLastPatientAwaitingRemoved)
+{
+    // first doctor queue
+    sut_.addAwaitingPatient(doctors_[0], patients_[3]);
+    sut_.addAwaitingPatient(doctors_[0], patients_[2]);
+    // second doctor queue
+    sut_.addAwaitingPatient(doctors_[1], patients_[2]);
+    sut_.addAwaitingPatient(doctors_[1], patients_[0]);
+
+    const auto doctors_awaiting_appointments_before = sut_.doctorsHavingQueue();
+    ASSERT_THAT(doctors_awaiting_appointments_before, UnorderedElementsAre(doctors_[0], doctors_[1]));
+
+    sut_.removeAwaitingPatient(doctors_[0], patients_[3]);
+    sut_.removeAwaitingPatient(doctors_[0], patients_[2]);
+    const auto doctors_awaiting_appointments_after_removing_first_doctor_patients = sut_.doctorsHavingQueue();
+
+    sut_.removeAwaitingPatient(doctors_[1], patients_[2]);
+    sut_.removeAwaitingPatient(doctors_[1], patients_[0]);
+    const auto doctors_awaiting_appointments_after_removing_second_doctor_patients = sut_.doctorsHavingQueue();
+
+    EXPECT_THAT(doctors_awaiting_appointments_after_removing_first_doctor_patients, ElementsAre(doctors_[1]));
+    EXPECT_TRUE(doctors_awaiting_appointments_after_removing_second_doctor_patients.empty());
+}
+
+class AwaitingPatientsShould : public ReserveQueueFixture
+{
+};
+
+TEST_F(AwaitingPatientsShould, ReturnEmptyQueueIfNoPatientForTreatmentQueued)
+{
+    sut_.addAwaitingPatient(Treatment::DentalSealants, patients_[0]);
+    sut_.addAwaitingPatient(Treatment::GumSurgery, patients_[1]);
+    ASSERT_FALSE(sut_.awaitingPatients(Treatment::DentalSealants).empty());
+    ASSERT_FALSE(sut_.awaitingPatients(Treatment::GumSurgery).empty());
+
+    EXPECT_TRUE(sut_.awaitingPatients(Treatment::ToothExtraction).empty());
+    EXPECT_TRUE(sut_.awaitingPatients(Treatment::OTHER).empty());
+}
+
+TEST_F(AwaitingPatientsShould, ReturnEmptyQueueIfNoPatientForDoctorQueued)
+{
+    sut_.addAwaitingPatient(doctors_[0], patients_[0]);
+    sut_.addAwaitingPatient(doctors_[1], patients_[1]);
+    ASSERT_FALSE(sut_.awaitingPatients(doctors_[0]).empty());
+    ASSERT_FALSE(sut_.awaitingPatients(doctors_[1]).empty());
+
+    EXPECT_TRUE(sut_.awaitingPatients(doctors_[2]).empty());
+}
+
+TEST_F(AwaitingPatientsShould, NotInsertTreatmentQueueKeyIfPatientForEmptyQueueRequested)
+{
+    const auto treatments_awaiting_before = sut_.treatmentsHavingQueue();
+
+    const auto patients_awaiting_treatment = sut_.awaitingPatients(Treatment::ToothExtraction);
+    const auto treatments_awaiting_after = sut_.treatmentsHavingQueue();
+
+    EXPECT_TRUE(patients_awaiting_treatment.empty());
+    EXPECT_TRUE(treatments_awaiting_before.empty());
+    EXPECT_TRUE(treatments_awaiting_after.empty());
+}
```

Comment:
Test seems not necessary. 
Such assertions could be added to EnqueuePatientsForTreatmentInCorrectOrder and other.

Reply:
IMHO I would like to leave the test separated.

I agree, that this checks might be put into another tests.
But I like one test testing one particular trait explicitly whenever possible.
And avoid complicating tests unnecessary.